src/app/shared/_guards/only-super-admin-role.guard.ts

```diff
@@ -0,0 +1,45 @@
+import { Injectable, OnDestroy } from '@angular/core';
+import { CanActivate, Router } from '@angular/router';
+import { CurrentUserService } from '@core/services/user/current-user.service';
+import { SnackBarService } from '@shared/services/snackbar.service';
+import { Subscription } from 'rxjs';
+
+@Injectable({
+	providedIn: 'root'
+})
+export class OnlySuperAdminRoleGuard implements CanActivate, OnDestroy {
+	curRoleId: string;
+	private subscription: Subscription;
+
+	constructor(
+		private currentUserService: CurrentUserService,
+		private routes: Router,
+		private snackBarService: SnackBarService
+	) {
+		this.subscription = this.currentUserService.currentUser$.subscribe(
+			curUser => (this.curRoleId = curUser.roleId)
+		);
+	}
+
+	canActivate() {
+		// '0' role id for super admin
+		if (this.curRoleId === '0') {
+			return true;
+		} else {
+			// this.routes.navigateByUrl('/main');
```

Comment:
Remove all commented code

Reply:
Why?

Reply:
This code commented because he block route to add, edit user if your role "not super admin".